File: 02 - ScoreBoard/frontend/src/types/league.ts
import type { CompetitionScope } from "./team";

export interface League {
  id: string;
  name: string;
  shortName: string;
  country: string;
  logoUrl: string | null;
  scope: CompetitionScope;
  /** Leagues not yet backed by real data show a "Coming soon" state. */
  isSupported: boolean;
  season: string;
}

export interface Standing {
  leagueId: string;
  position: number;
  teamId: string;
  played: number;
  won: number;
  drawn: number;
  lost: number;
  goalsFor: number;
  goalsAgainst: number;
  goalDifference: number;
  points: number;
  form?: MatchResultLetter[];
}

export type MatchResultLetter = "W" | "D" | "L";

File: 02 - ScoreBoard/backend/app/models/league.py
from typing import List, Literal, Optional

from .common import CamelModel
from .team import CompetitionScope

MatchResultLetter = Literal["W", "D", "L"]


class League(CamelModel):
    id: str
    name: str
    short_name: str
    country: str
    logo_url: Optional[str] = None
    scope: CompetitionScope
    is_supported: bool
    season: str


class Standing(CamelModel):
    league_id: str
    position: int
    team_id: str
    played: int
    won: int
    drawn: int
    lost: int
    goals_for: int
    goals_against: int
    goal_difference: int
    points: int
    form: Optional[List[MatchResultLetter]] = None
